lib/screens/signup_page.dart

```diff
@@ -56,107 +61,183 @@ class _SignupPageState extends State<SignupPage> {
+                  return 'Please enter a valid email address.';
                 }
                 _userEmail = value.trim();
                 return null;
               },
-              textAlign: TextAlign.center,
             ),
+            SizedBox(height: 20.0),
             TextFormField(
-              obscureText: true,
-              decoration: const InputDecoration(
-                hintText: 'Password',
+              obscureText: _isHiddenPassword,
+              decoration: InputDecoration(
+                labelText: 'Password',
+                filled: true,
+                fillColor: Color(0xF6F6F6F6),
+                enabledBorder: OutlineInputBorder(
+                  borderSide: BorderSide(
+                    color: Colors.grey[200],
+                  ),
+                ),
+                prefixIcon: Icon(Icons.lock),
+                suffixIcon: IconButton(
+                  onPressed: () {
+                    setState(() {
+                      _isHiddenPassword = !_isHiddenPassword;
+                    });
+                  },
+                  icon: Icon(
+                    _isHiddenPassword ? Icons.visibility : Icons.visibility_off,
+                  ),
+                ),
               ),
               validator: (value) {
                 if (value.isEmpty) {
-                  return 'Password Needed';
+                  return 'Password required.';
                 }
                 _password = value;
                 return null;
               },
-              textAlign: TextAlign.center,
             ),
+            SizedBox(height: 20.0),
             TextFormField(
-              obscureText: true,
-              decoration: const InputDecoration(
-                hintText: 'Confirm Password',
+              obscureText: _isHiddenPassword,
+              decoration: InputDecoration(
+                labelText: 'Confirm Password',
+                filled: true,
+                fillColor: Color(0xF6F6F6F6),
+                enabledBorder: OutlineInputBorder(
+                  borderSide: BorderSide(
+                    color: Colors.grey[200],
+                  ),
+                ),
+                prefixIcon: Icon(Icons.lock),
+                suffixIcon: IconButton(
+                  onPressed: () {
+                    setState(() {
+                      _isHiddenPassword = !_isHiddenPassword;
+                    });
+                  },
+                  icon: Icon(
+                    _isHiddenPassword ? Icons.visibility : Icons.visibility_off,
+                  ),
+                ),
               ),
               validator: (value) {
                 if (value.isEmpty) {
-                  return 'Retype Password';
+                  return 'Password required.';
                 } else if (_password != value) {
-                  return 'Password does not match';
+                  return 'Passwords do not match.';
                 }
```

Comment:
Here I believe another case and message are needed: "Password should be at least 6 characters". I am not sure, but I believe it can be added to firebase exception handler file